js/src/edit-free-campaign/hasUnsavedShippingRages.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * External dependencies
+ */
+import { isEqual, differenceWith, at } from 'lodash';
+
+/**
+ * @typedef {import('.~/data/actions').ShippingRate} ShippingRate
+ */
+
+/**
+ * This function compares the key fields of two shipping rate arrays
+ * and returns whether there are any unsaved shipping changes.
+ *
+ * - Order:     The order of two arrays does not need to be the same.
+ * - ID:        The `id` is not compared.
+ *              On the UI, if a rate was deleted and then added,
+ *              then the `id` would not exist.
+ * - Threshold: If `options` or `options.free_shipping_threshold` does not exist,
+ *              `free_shipping_threshold` is treated as undefined.
+ *
+ * @param  {Array<ShippingRate>} rates The shipping rates to be compared.
+ * @param  {Array<ShippingRate>} savedRates The saved shipping rates received from API.
+ * @return {boolean} Whether there are any unsaved shipping changes.
+ */
+export default function hasUnsavedShippingRages( rates, savedRates ) {
```

Comment:
💅 💅 💅  Please chill, no rages. 🤣 

Note: you got to rename this in your editor, and rename the file name, and also make the necessary changes in your PR description. 😄 

```suggestion
export default function hasUnsavedShippingRates( rates, savedRates ) {
```

Reply:
Oh no 🤦‍♂️  Thanks for catching this! 😂 

Fixed in https://github.com/woocommerce/google-listings-and-ads/pull/1307/commits/d38dab59344616eacccc48422473eee434be1421.